fast64_internal/f3d/f3d_material.py

```diff
@@ -836,6 +837,39 @@ def ui_fog(self, f3dMat, inputCol, showCheckBox):
 			#fogGroup.enabled = not f3dMat.use_global_fog
 			#inputGroup.box().label(text = 'NOTE: Fog will break with draw layer overrides.')
 
+	
+	def ui_cel_shading(self, material, layout):
+		nodes = material.node_tree.nodes
+		inputGroup = layout.box().column()
+
+		if material.mat_ver > 3:
+			inputGroup.prop(material.f3d_mat, "do_cel_shading")
+			if material.f3d_mat.do_cel_shading:
+				cel = material.f3d_mat.cel_shading
+				inputGroup.prop(cel, "solid")
+				for l in cel.levels:
+					box = inputGroup.box().column()
+					r = box.row()
+					r.prop(l, "inverse")
+					r.prop(l, "lighten")
+					r = box.row().split(factor=0.25, align=True)
+					r.label(text = 'Color' if l.lighten else 'Black')
+					r = r.split(factor=0.66, align=True)
+					r.prop(l, "fade")
+					r.label(text = 'White' if l.lighten else 'Color')
+					r = box.row().split(factor=0.3, align=True)
+					r.label(text = 'Write if ' + ('<=' if l.inverse else '>='))
```

Comment:
I think just using the term "threshold" here would be easier to understand, something simple like "Light Threshold"?

Reply:
The term "threshold" is in the box to the right of this (the next prop). It says "Write if >= Threshold 128" or "Write if <= Threshold 128" or whatever. But maybe "draw if" or "fill if" is better than "write if".